mentat/code_map.py

```diff
@@ -25,17 +32,32 @@ def _get_code_map(root: Path, file_path: Path, exclude_signatures: bool = False)
     else:
         ctags_cmd_args.append("--fields=+S")
     ctags_cmd = ["ctags", *ctags_cmd_args, str(Path(root).joinpath(file_path))]
-    output = subprocess.check_output(ctags_cmd, stderr=subprocess.PIPE, text=True)
-    output_lines = output.splitlines()
+
+    process = await asyncio.create_subprocess_exec(
+        *ctags_cmd,
+        stdout=asyncio.subprocess.PIPE,
+        stderr=asyncio.subprocess.PIPE,
+        preexec_fn=os.setpgrp,
+    )
+    stdout, stderr = await process.communicate()
+
+    if process.returncode == 0:
+        output = stdout.decode("utf-8").strip() if stdout else ""
+        output_lines = output.splitlines()
+    else:
+        error_output = stderr.decode("utf-8").strip() if stderr else ""
+        raise Exception(f"Command failed with error: {error_output}")
```

Comment:
should we make a helper function that runs async subprocess calls?

Reply:
yea that's a good idea